src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/Ecma/MetadataTable.cs

```diff
@@ -26,12 +25,12 @@ public MetadataTable(int count)
             _table = new T?[count];
         }
 
-        public T GetOrAdd(EntityHandle handle, C context, Func<EntityHandle, C, T> factory)
+        public T GetOrAdd(EntityHandle handle, EcmaModule context, Func<EntityHandle, EcmaModule, T> factory)
         {
             Debug.Assert(!handle.IsNil);
             Debug.Assert(factory != null);
 
-            int index = handle.GetToken().GetTokenRowNumber() - 1;
```

Comment:
> System.Reflection handles WinMDs fine, because it knows about virtual AssemblyRefs

I assume you mean `System.Reflection.Metadata`, not `System.Reflection` via the reader option `MetadataReaderOptions.ApplyWindowsRuntimeProjections` and the resulting `MetadataKind.WindowsMetadata` enum value.